src/Uno.Extensions.Reactive.UI/Presentation/Bindings/Collections/Services/SelectionService.cs

```diff
@@ -50,7 +50,7 @@ public IReadOnlyList<ItemIndexRange> GetSelectedRanges()
 	/// <inheritdoc />
 	public void SelectRange(ItemIndexRange itemIndexRange)
 	{
-		if (Update((info, removed) => info.Add(removed), new SelectionIndexRange((uint)itemIndexRange.FirstIndex, (uint)itemIndexRange.LastIndex)))
+		if (Update((info, added) => info.Add(added), new SelectionIndexRange((uint)itemIndexRange.FirstIndex, itemIndexRange.Length)))
```

Comment:
SelectionIndexRange could use a static `From(ItemIndexRange)` method or implicit operator

Reply:
Yes but no, the `ItemIndexRange` is in the the UI namespace while the `SelectionIndexRange` is in "core" library. Actually I was planning to use `ItemIndexRange` itself but I was forced to create my own one.